test/shaping/run-tests.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import sys, os, subprocess
+
+def cmd(command):
+	p = subprocess.Popen (
+		command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```

Comment:
Do you actually need shell=True? It spawns a sub shell. Only use it if you really use commands that only work inside a shell

Reply:
It is in use for "../.." part, as this is a replacement for a bash script, hmm let me ask @behdad, is this script in use on projects outside harfbuzz? Guessed that because of smilingly unused `--reference` defined on the script.

Reply:
you can do `os.path.join("..", "..")`

even the two dots have their portable alias in `os.path.pardir`

Reply:
Done

Reply:
> It is in use for "../.." part, as this is a replacement for a bash script, hmm let me ask @behdad, is this script in use on projects outside harfbuzz? Guessed that because of smilingly unused --reference defined on the script.

Script is not used outside, except that one can use --reference to regenerate the expected-output file, when things change in harfbuzz, eg. glyph-names are implemented in ot-fonts or other things that invalidate the output.  Or just to write test manually and generate expected output.